XouDevSpec/Classes/features/RegisterScreen/ViewModel/RegisterViewModel.swift

```diff
@@ -0,0 +1,36 @@
+//
+//  RegisterViewModel.swift
+//  XouDevSpec
+//
+//  Created by Xander Schoeman on 2020/04/27.
+//
+
+import Foundation
+
+public protocol RegisterViewModelProtocol {
+    func registerAUser(withUser: User)
+}
+
+public class RegisterViewModel {
+    weak var view: RegisterViewProtocol?
+    var Repo: XouBaseApiServiceProtocol
+    
+    public init(with view: RegisterViewProtocol, repo: XouBaseApiServiceProtocol) {
+        self.view = view
+        self.Repo = repo
+    }
+}
+
+extension RegisterViewModel: RegisterViewModelProtocol {
+    
+    public func registerAUser(withUser: User){
+        Repo.registerUser(theUser: withUser, completetionHandler: { result in
+            switch result {
+            case .success(let user):
+                print("Success the user: \(user.username) has been registered")
+            case .failure(let error):
+                print("An error has occured: \(error)")
+            }
+        })
```

Comment:
The user has no way of knowing if the success or failure triggered. In fact, it seems the view has no way of knowing if something happened at all.

Reply:
You would use your ViewProtocol over here to tell the view something happened

Reply:
This needs to be fixed as it does create dead code in RegisterViewProtocol that shouldn't be dead

Reply:
On it thanks.